extensions/camel/camel-core/runtime/src/main/java/io/quarkus/camel/core/runtime/CamelRuntime.java

```diff
@@ -71,7 +72,13 @@ public void doInit() {
             RuntimeSupport.bindProperties(properties, context, PFX_CAMEL_CONTEXT);
 
             context.setLoadTypeConverters(false);
-            context.getModelJAXBContextFactory().newJAXBContext();
+
+            // The creation of the JAXB context is very time consuming, so always prepare it
+            // when running in native mode, but lazy create it in java mode so that we don't
+            // waste time if using java routes
+            if (ImageInfo.inImageBuildtimeCode()) {
```

Comment:
Assuming you parse all routes at build time, wouldn't you want JAXB initialized lazily in that same phase as well?

Or is this relating to Camel-K possibly not building any route at build time, then add them "dynamically" after it was all compiled to native already?

Reply:
Not sure to understand.  Camel routes can be loaded using java code or by parsing the XML and using jaxb.  This can happen either at build time or at runtime.
What I'd like to avoid is to create the JAXB context at runtime when there's no XML routes to load.  I agree that if routes are loaded at build time using java, it will initialize the JAXB context for nothing, but this is done at build time and there's currently no way to completely get rid of the jaxb code afaik, so I don't think that's an issue (although I'd like to investigate being able to trim down camel apps to remove all xml support if possible).

Reply:
ok thanks I understand now, sorry for the confusion.

I think we'd normally do such a thing with a different build item, but I won't slow you down.